apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -376,12 +376,14 @@ export const addStudents = studentIds => {
 };
 
 // Creates a new RowType.NEW_STUDENT for each name in the array.
-export const addMultipleAddRows = studentNames => {
+export const addMultipleAddRows = studentDataArray => {
   return (dispatch, getState) => {
     let studentData = {};
-    for (let i = 0; i < studentNames.length; i++) {
-      // Do not add rows with no name
-      if (studentNames[i] === '') {
+    for (let i = 0; i < studentDataArray.length; i++) {
+      const {name, familyName} = studentDataArray[i];
+
+      // Do not add rows with no display name
+      if (name === '') {
```

Comment:
A bit more of a functional, succinct way to do this:

```suggestion
    const studentData = studentDataArray
      .filter(data => data.name)
      .reduce((accumulator, data) => {
        const newId = addRowIdCounter--;

        return {
          ...accumulator,
          [newId]: {
            ...blankNewStudentRow,
            name: data.name,
            familyName: data.familyName,
            id: newId,
          },
        };
      }, {});
```